diff --git a/gnutls.c b/gnutls.c
index baf1cd1..ef32c52 100644
--- a/gnutls.c
+++ b/gnutls.c
@@ -1971,55 +1971,18 @@ static int verify_peer(gnutls_session_t session)
 		   why we don't just set a bit for that too. */
 		reason = _("signature verification failed");
 
-	if (reason)
-		goto done;
-
-	if (!gnutls_x509_crt_check_hostname(cert, vpninfo->hostname)) {
-		int i, ret;
-		unsigned char addrbuf[sizeof(struct in6_addr)];
-		unsigned char certaddr[sizeof(struct in6_addr)];
-		size_t addrlen = 0, certaddrlen;
-
-		/* gnutls_x509_crt_check_hostname() doesn't cope with IPv6 literals
-		   in URI form with surrounding [] so we must check for ourselves. */
-		if (vpninfo->hostname[0] == '[' &&
-		    vpninfo->hostname[strlen(vpninfo->hostname)-1] == ']') {
-			char *p = &vpninfo->hostname[strlen(vpninfo->hostname)-1];
-			*p = 0;
-			if (inet_pton(AF_INET6, vpninfo->hostname + 1, addrbuf) > 0)
-				addrlen = 16;
-			*p = ']';
-		}
-#if GNUTLS_VERSION_NUMBER < 0x030306
-		/* And before 3.3.6 it didn't check IP addresses at all. */
-		else if (inet_pton(AF_INET, vpninfo->hostname, addrbuf) > 0)
-			addrlen = 4;
-		else if (inet_pton(AF_INET6, vpninfo->hostname, addrbuf) > 0)
-			addrlen = 16;
-#endif
-
-		if (!addrlen) {
-			/* vpninfo->hostname was not a bare IP address. Nothing to do */
-			goto badhost;
-		}
+	if (reason) {
+		vpn_progress(vpninfo, PRG_INFO,
+			     _("Server certificate verify failed: %s (ignoring)\n"),
+			     reason);
+		reason = NULL;
+	}
 
-		for (i = 0; ; i++) {
-			certaddrlen = sizeof(certaddr);
-			ret = gnutls_x509_crt_get_subject_alt_name(cert, i, certaddr,
-								   &certaddrlen, NULL);
-			/* If this happens, it wasn't an IP address. */
-			if (ret == GNUTLS_E_SHORT_MEMORY_BUFFER)
-				continue;
-			if (ret < 0)
-				break;
-			if (ret != GNUTLS_SAN_IPADDRESS)
-				continue;
-			if (certaddrlen == addrlen && !memcmp(addrbuf, certaddr, addrlen))
-				goto done;
+		/* Skip hostname mismatch check - force connection */
+		if (!gnutls_x509_crt_check_hostname(cert, vpninfo->hostname)) {
+			vpn_progress(vpninfo, PRG_INFO,
+			     _("certificate does not match hostname (ignoring)\n"));
 		}
-	badhost:
-		reason = _("certificate does not match hostname");
-	}
  done:
 	if (reason) {
 		vpn_progress(vpninfo, PRG_INFO,
